events/system/interactionCreate.js

```diff
@@ -29,6 +30,24 @@ export default {
         });
       }
 
+      if (command.guards?.length) {
+        for (const guardName of command.guards) {
+          const guard = guardMap[guardName];
+
+          if (!guard) {
+            console.error(`Unknown guard: ${guardName}`);
+            await interaction.reply({
+              content: "Configuration error.",
+              flags: MessageFlags.Ephemeral,
+            });
+            return;
+          }
+
+          const passed = await guard(interaction);
+          if (!passed) return;
+        }
+      }
```

Comment:
We should import the guards into the commands and setup an array of guards in the preferred order, rather than allow for the possibility of typos.
```suggestion
      if (command.guards?.length) {
        for (const guard of command.guards) {
          const result = await guard(interaction);

          if (!result) {
            await interaction.reply({
              content: "You don't meet the requirements for this command.",
              ephemeral: true,
            });
            return;
          }

          if (result && result.passed === false) {
            await interaction.reply({
              content: result.reason || "Access denied.",
              ephemeral: true,
            });
            return;
          }
        }
      }
```